service/prometheus/service.go

```diff
@@ -115,16 +117,21 @@ func (s *Service) isReloadRateLimited() bool {
 	timeSinceLastReload := time.Since(s.lastReloadTime)
 
 	if timeSinceLastReload < s.minimumReloadTime {
-		s.logger.Log("debug", "ignoring reload request, only %t since last reload, minimum time between is %t", timeSinceLastReload, s.minimumReloadTime)
+		s.logger.Log("debug", fmt.Sprintf("ignoring reload request, only %s since last reload, minimum time between is %s", timeSinceLastReload, s.minimumReloadTime))
 		configurationReloadIgnoredCount.Inc()
 
 		return true
 	}
-	s.lastReloadTime = time.Now()
 
 	return false
 }
 
+func (s *Service) RequestReload() {
+	s.logger.Log("debug", "reload requested")
+
+	s.isReloadRequested = true
```

Comment:
As we discussed this should use a mutex to lock and unlock afterwards.

Reply:
I don't think a mutex is necessary, actually. We do all this foo sequentially.

Reply:
I hope you get woken up because of this at some point.